src/main/java/org/springframework/data/config/ConfigurationUtils.java

```diff
@@ -88,16 +84,14 @@ public static ClassLoader getRequiredClassLoader(ResourceLoader resourceLoader)
 	 * @return
 	 * @throws IllegalArgumentException if the given {@link BeanDefinition} does not contain a bean class name.
 	 */
-	public static String getRequiredBeanClassName(BeanDefinition beanDefinition) {
+	static String getRequiredBeanClassName(BeanDefinition beanDefinition) {
 
 		Assert.notNull(beanDefinition, "BeanDefinition must not be null!");
 
 		String result = beanDefinition.getBeanClassName();
 
-		if (result == null) {
-			throw new IllegalArgumentException(
-					String.format("Could not obtain required bean class name from BeanDefinition!", beanDefinition));
-		}
+		Assert.notNull(result, String.format("Could not obtain required bean class name from BeanDefinition [%s]!",
```

Comment:
This change has performance impact as `String.format(…)` is invoked regardless of whether it's required to. We should either stick with the previous variant or use a message supplier. There are also occurrences of similar changes.